code/game/machinery/computer/security.dm

```diff
@@ -2,311 +2,193 @@
 #define SEC_DATA_MAINT	2	// Records maintenance
 #define SEC_DATA_RECORD	3	// Record
 
-/obj/machinery/computer/secure_data//TODO:SANITY
+#define SEC_FIELD(N, V, E, LB) list(field = N, value = V, edit = E, line_break = LB)
+
+/obj/machinery/computer/secure_data
 	name = "security records"
 	desc = "Used to view and edit personnel's security records."
 	icon_keyboard = "security_key"
 	icon_screen = "security"
-	req_one_access = list(ACCESS_SECURITY, ACCESS_FORENSICS_LOCKERS)
 	circuit = /obj/item/circuitboard/secure_data
-	var/obj/item/card/id/scan = null
-	var/authenticated = null
-	var/rank = null
-	var/list/authcard_access = list()
-	var/screen = null
-	var/datum/data/record/active1 = null
-	var/datum/data/record/active2 = null
-	var/temp = null
-	var/printing = null
-	//Sorting Variables
-	var/sortBy = "name"
-	var/order = 1 // -1 = Descending - 1 = Ascending
+	/// The inserted ID card's access list.
+	var/list/inserted_id_access = null
+	/// The current page being viewed.
+	var/current_page = SEC_DATA_R_LIST
```

Comment:
So are you handling tabs server side instead of client side?

Reply:
Server-side because it dictates which data is sent (list of records, specific record..)